datarepo-clienttests/src/main/java/common/utils/KubernetesClientUtils.java

```diff
@@ -370,6 +371,7 @@ private static long getApiReadyPods(V1Deployment deployment) throws ApiException
                 pod ->
                     deploymentComponentLabel.equals(
                             pod.getMetadata().getLabels().get(componentLabel))
+                        && pod.getStatus().getContainerStatuses() != null
```

Comment:
Not sure how bulletproof this code needs to be but `getLabels()` and `get()` can also return null here.

Reply:
I think if pods don't have metadata or labels, then something is wrong/unexpected with the deployment and we have to go investigate manually. So I'm not sure it really improves error reporting to add more null checks. But I could go either way on this. Happy to hear other's thoughts/preferences.

Reply:
IMHO in a tool like this, an NPE is as good as a null test and throwing some other exception.